### PR TITLE
samples/bpf: xdp_ddos01 add support for port blacklisting

### PR DESCRIPTION
**Pull request by gospo**
This adds support for keeping an array-based blacklist of ports.
Perf report indicates that when bpf JITing is enabled in the kernel
there is a 3-4% decrease in cpu samples (measured with perf) to using
this over the new inlined bpf hash maps for storing port blacklist data.
Perf report indicates performance is roughly equal comparing this method
to bpf mash maps when bpf JITing is disabled in the kernel.

Signed-off-by: Andy Gospodarek <gospo@broadcom.com>

### CODE REVIEW

**Comment by gospo** [comment]
We should talk more about this if you have any questions.  LOTS of testing went into this as the optimal solution -- I can explain what I tested in more detail if you like.

File: kernel/samples/bpf/xdp_ddos01_blacklist_cmdline.c [hunk 1]
```diff
@@ -197,21 +210,51 @@ static void blacklist_print_ip(__u32 ip, __u64 count)
...
+}
+
 static void blacklist_list_all(int fd)
 {
 	__u32 key = 0, next_key;
 	__u64 value;
 
 	printf("{\n");
 	while (bpf_map_get_next_key(fd, &key, &next_key) == 0) {
-		printf("%s", key ? "," : " ");
```
Comment (netoptimizer) [inline kernel/samples/bpf/xdp_ddos01_blacklist_cmdline.c]: Why did you remove this?
(Notice this allowed --list to be parsed as JSON)

**Comment by netoptimizer** [comment]
I'm just going to pull in this, but I don't have time test it at the moment...

File: kernel/samples/bpf/xdp_ddos01_blacklist_cmdline.c (hunk 1 above)
Comment (gospo) [inline kernel/samples/bpf/xdp_ddos01_blacklist_cmdline.c]: Sorry that was accidental.

I'll add back as I also have some thoughts on adding a --port and --address args to --list so that the output is JSON for both.
